crates/wast/src/resolve/mod.rs

```diff
@@ -73,14 +97,26 @@ pub fn resolve<'a>(module: &mut Module<'a>) -> Result<Names<'a>, Error> {
 
     // Expand all `TypeUse` annotations so all necessary `type` nodes are
     // present in the AST.
-    types::expand(fields);
+    types::expand(fields, &mut gensym);
 
     // Perform name resolution over all `Index` items to resolve them all to
     // indices instead of symbolic names.
-    let resolver = names::resolve(module.id, fields)?;
+    let resolver = names::resolve(fields)?;
     Ok(Names { resolver })
 }
 
+pub fn resolve_component<'a>(component: &mut Component<'a>) -> Result<(), Error> {
+    let fields = match &mut component.kind {
+        ComponentKind::Text(fields) => fields,
+        _ => return Ok(Default::default()),
+    };
+
```

Comment:
I think one thing this is missing is the de-inlining/elaboration, right? 

Or, more generally, would you be up for making a checklist of remaining TODO items after this?